js/coindcx.js

```diff
@@ -0,0 +1,596 @@
+        return this.parseTrades (response, market, since, limit);
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = 500, params = {}) {
+        // https://coindcx-official.github.io/rest-api/?javascript#account-trade-history
+        await this.loadMarkets ();
+        const request = {
+            'timestamp': this.milliseconds (),
+            'limit': limit,
+        };
+        const response = await this.privatePostExchangeV1OrdersTradeHistory (this.extend (request, params));
+        return this.parseTrades (response, undefined, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        const timestamp = this.safeInteger2 (trade, 't', 'timestamp');
+        let symbol = undefined;
+        if (market === undefined) {
+            const marketId = this.safeString2 (trade, 's', 'symbol');
+            market = this.safeValue (this.markets_by_id, marketId);
+        }
+        if (market !== undefined) {
+            symbol = market['symbol'];
+        }
+        let takerOrMaker = undefined;
+        if ('m' in trade) {
+            takerOrMaker = trade['m'] ? 'maker' : 'taker';
+        }
+        const price = this.safeFloat2 (trade, 'p', 'price');
+        const amount = this.safeFloat2 (trade, 'q', 'quantity');
+        return {
+            'id': this.safeString (trade, 'id'),
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'order': undefined,
+            'type': undefined,
+            'takerOrMaker': takerOrMaker,
+            'side': this.safeString (trade, 'side'),
+            'price': price,
+            'amount': amount,
+            'cost': price * amount,
+            'fee': this.safeFloat (trade, 'fee_amount'),
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        // https://coindcx-official.github.io/rest-api/?shell#order-book
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrderBook requires a `symbol` argument');
+        }
+        await this.loadMarkets ();
+        if (!(symbol in this.markets)) {
+            throw new ExchangeError (this.id + ' has no symbol ' + symbol);
+        }
+        const market = this.market (symbol);
+        const marketInfo = this.safeValue (market, 'info');
+        const coindcxPair = this.safeString (marketInfo, 'pair');
+        if (coindcxPair === undefined) {
+            throw new ExchangeError (this.id + ' has no pair (look at market\'s info) value for ' + symbol);
+        }
+        const request = {
+            'pair': coindcxPair,
+        };
+        const response = await this.publicGetMarketDataOrderbook (this.extend (request, params));
+        return this.parseOrderBookData (response);
+    }
+
+    parseOrderBookData (orderBook) {
```

Comment:
This method is redundant, see how it is done in the other exchanges with `parseOrderBook()`.

Reply:
I made custom wrapper because the exchange gives the data in another format: 
`
{
  "bids": {
    "price": "amount",
    "price": "amount",
    ...
  },
  "asks": {
    "price": "amount",
    "price": "amount",
    ...
  }
}
`
instead of `{"bids": [['price', 'amount'], ['price', 'amount']...]}`
Or I should parse data into the format you use and use your wrapper?

Reply:
@eiamei 

> I made custom wrapper because the exchange gives the data in another format:

ok

> Or I should parse data into the format you use and use your wrapper?

In this case, I'd rather override just the `parseBidsAsks`. We definitely don't want to do a double conversion, so we should either leave it as is then, or rewrite it a bit to still use `parseOrderBook` with a custom `parseBidsAsks` override.

Reply:
ok, will do this

Reply:
@eiamei thx so much! I'll do a more thorough review once again soon and add more comments in between the lines soon, if I find anything else.